homeassistant/components/wake_on_lan/switch.py

```diff
@@ -114,31 +118,38 @@ def turn_on(self, **kwargs):
 
         wakeonlan.send_magic_packet(self._mac_address, **service_kwargs)
 
+        if self._is_dummy:
+            self._state = True
+
     def turn_off(self, **kwargs):
         """Turn the device off if an off action is present."""
         if self._off_script is not None:
             self._off_script.run(context=self._context)
 
+        if self._is_dummy:
+            self._state = False
```

Comment:
This would set the state regardless of the presence (and execution) of an off action. I don't this expected behaviour.

Reply:
I could check for an off action, but then the switch will never turn off in that case except for (I believe) a restart. I'm not sure that that would be better.

Reply:
```suggestion
        if self._is_dummy:
            self._state = False
            self.async_write_ha_state()
```